kroxylicious/src/main/java/io/kroxylicious/proxy/internal/MeterRegistries.java

```diff
@@ -80,4 +81,13 @@ private static Object tagNames(Meter.Id id1) {
     public Optional<PrometheusMeterRegistry> maybePrometheusMeterRegistry() {
         return Optional.ofNullable(prometheusMeterRegistry);
     }
+
+    @Override
+    public void close() {
+        hooks.forEach(MicrometerConfigurationHook::close);
+        // remove the meters we contributed to the global registry.
+        var copy = List.copyOf(prometheusMeterRegistry.getMeters());
+        copy.forEach(Metrics.globalRegistry::remove);
```

Comment:
Shouldn't we have a null check here? Or use `Optional.ifPresent`? 

Is there a particular reason for the defensive copy? I don't suppose the allocation cost really matters during close...

Reply:
`prometheusMeterRegistry.getMeters()` cannot return null. It is documented as `The set of registered meters.`

I used a defensive copy to avoid the possibility of `ConcurrentModificationException`.  However, `prometheusMeterRegistry.getMeters()` actually produces an unmodifiable copy, so my wrap technically superfluous.  That said, the javadoc doesn't actually guarantee that's the case and `MeterRegistry#forEachMeter is backed by the live structure. Their API isn't too consistent.  That makes me want to keep the copy.

Reply:
The registry could be `null` iirc

Reply:
prometheusMeterRegistry is assigned in the ctor and is not null.